src/Spotify/SpotifyAPITypes.h

```diff
@@ -211,15 +211,15 @@ class SpotifyPaging {
   SpotifyPaging() = default;
   SpotifyPaging(nlohmann::json const &pagingJson);
 
-  std::vector<Track> const &getTracks();
-  std::vector<Artist> const &getArtists();
-  std::vector<Album> const &getAlbums();
-  std::string const &getHref();
-  int getLimit();
-  std::string const &getNext();
-  int getOffset();
-  std::string const &getPrevious();
-  int getTotal();
+  std::vector<Track> const &getTracks() const;
```

Comment:
I dont know if thread safety is needed here, but be careful when returning references.

Reply:
i used the references here, because in my opinion the object must exist anyway if you access to the members (same as i would make the members public)
you mean the thread safety in refreshAccess token?

Reply:
the comment was just meant to be a reminder. as long as you do not access the same instance from multiple threads it's fine

Reply:
ah ok